cuda/matrix/csr_kernels.cu

```diff
@@ -895,7 +1105,31 @@ void conj_transpose(std::shared_ptr<const CudaExecutor> exec,
             orig->get_size()[1], orig->get_num_stored_elements(),
             orig->get_const_values(), orig->get_const_row_ptrs(),
             orig->get_const_col_idxs(), trans->get_values(),
-            trans->get_col_idxs(), trans->get_row_ptrs(), copyValues, idxBase);
+            trans->get_row_ptrs(), trans->get_col_idxs(), copyValues, idxBase);
+#else  // CUDA_VERSION >= 11000
+        cudaDataType_t cu_value =
+            gko::kernels::cuda::cuda_data_type<ValueType>();
+        cusparseAction_t copyValues = CUSPARSE_ACTION_NUMERIC;
+        cusparseIndexBase_t idxBase = CUSPARSE_INDEX_BASE_ZERO;
+        cusparseCsr2CscAlg_t alg = CUSPARSE_CSR2CSC_ALG2;
```

Comment:
Important: We should always keep this at ALG1, since ALG2 doesn't guarantee any ordering of the columns. This could silently break a lot of algorithms! Can you maybe add a test for Csr::transpose (if we don't already have one) that tests that `is_sorted_by_column_index()` is true?
```suggestion
        cusparseCsr2CscAlg_t alg = CUSPARSE_CSR2CSC_ALG1;
```